langchain4j-oracle/src/main/java/dev/langchain4j/data/document/loader/oracle/OracleDocumentLoader.java

```diff
@@ -0,0 +1,170 @@
+package dev.langchain4j.data.document.loader.oracle;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.Metadata;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OracleDocumentLoader {
+
+    private static final Logger log = LoggerFactory.getLogger(OracleDocumentLoader.class);
+
+    private final Connection conn;
+
+    public OracleDocumentLoader(Connection conn) {
+        this.conn = conn;
+    }
+
+    public List<Document> loadDocuments(String pref) {
+        List<Document> documents = new ArrayList<>();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode rootNode = mapper.readTree(pref);
+            JsonNode fileNode = rootNode.path("file");
+            JsonNode dirNode = rootNode.path("dir");
+            JsonNode ownerNode = rootNode.path("owner");
+            JsonNode tableNode = rootNode.path("tablename");
+            JsonNode colNode = rootNode.path("colname");
+
+            if (fileNode.textValue() != null) {
+                String filename = fileNode.textValue();
+                Document doc = loadDocument(filename, pref);
+                if (doc != null) {
+                    documents.add(doc);
+                }
+            } else if (dirNode.textValue() != null) {
+                String dir = dirNode.textValue();
+                Path root = Paths.get(dir);
+                Files.walk(root).forEach(path -> {
+                    if (path.toFile().isFile()) {
+                        Document doc = loadDocument(path.toFile().toString(), pref);
+                        if (doc != null) {
+                            documents.add(doc);
+                        }
+                    }
+                });
+            } else if (colNode.textValue() != null) {
+                String column = colNode.textValue();
+
+                String table = tableNode.textValue();
+                String owner = ownerNode.textValue();
+                if (table == null) {
+                    throw new RuntimeException("Missing table in preference");
+                }
+                if (owner == null) {
+                    throw new RuntimeException("Missing owner in preference");
+                }
+
+                documents.addAll(loadDocuments(owner, table, column, pref));
+            } else {
+                throw new RuntimeException("Invalid preference");
```

Comment:
I think we should add more information in here, and it's probably more a InvalidParameterException: https://docs.oracle.com/javase/8/docs/api/java/security/InvalidParameterException.html